### PR TITLE
Add integ tests for model APIs

### PR DESCRIPTION
**Pull request by Zhangxunmt**
Signed-off-by: Xun Zhang <xunzh@amazon.com>

### Description
Add integration tests for all model APIs. Rebased on PR 167.
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: plugin/src/test/java/org/opensearch/ml/rest/MLCommonsRestTestCase.java [hunk 1]
```diff
@@ -283,7 +284,16 @@ protected void validateStats(
         }
         assertEquals(expectedTotalFailureCount, totalFailureCount);
         assertEquals(expectedTotalAlgoFailureCount, totalAlgoFailureCount);
-        assertEquals(expectedTotalRequestCount, totalRequestCount);
+        // ToDo: this line makes this test flaky as other tests makes the request count not predictable
+        // assertEquals(expectedTotalRequestCount, totalRequestCount);
```
Comment (ylwu-amzn) [inline plugin/src/test/java/org/opensearch/ml/rest/MLCommonsRestTestCase.java:288]: I think it's ok to remove this line, multiple tests against same cluster may have different total requests based on their execution sequence.

Comment (ylwu-amzn) [inline plugin/src/test/java/org/opensearch/ml/rest/MLCommonsRestTestCase.java:288]: I think it's ok to remove this line, multiple tests against same cluster may have different total requests based on their execution sequence.

File: plugin/src/test/java/org/opensearch/ml/rest/RestMLDeleteModelActionIT.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import org.apache.http.HttpEntity;
+import org.opensearch.client.Response;
+import org.opensearch.client.ResponseException;
+import org.opensearch.ml.utils.TestHelper;
+import org.opensearch.rest.RestStatus;
+
+public class RestMLDeleteModelActionIT extends MLCommonsRestTestCase {
+
+    public void testDeleteModelAPI_EmptyResources() throws Exception {
+        TestHelper
```
Comment (ylwu-amzn) [inline plugin/src/test/java/org/opensearch/ml/rest/RestMLDeleteModelActionIT.java:20]: To keep consistent with other code, how about using `ExpectedException` ?
```
    @Rule
    public ExpectedException exceptionRule = ExpectedException.none();

    public void testGetModelAPI_EmptyResources() throws IOException {
        exceptionRule.expect(ResponseException.class);
        exceptionRule.expectMessage("index_not_found_exception");
        TestHelper.makeRequest(client(), "DELETE", "/_plugins/_ml/models/111222333", null, "", null);
    }
```

Comment (ylwu-amzn) [inline plugin/src/test/java/org/opensearch/ml/rest/RestMLDeleteModelActionIT.java:20]: To keep consistent with other code, how about using `ExpectedException` ?
```
    @Rule
    public ExpectedException exceptionRule = ExpectedException.none();

    public void testGetModelAPI_EmptyResources() throws IOException {
        exceptionRule.expect(ResponseException.class);
        exceptionRule.expectMessage("index_not_found_exception");
        TestHelper.makeRequest(client(), "DELETE", "/_plugins/_ml/models/111222333", null, "", null);
    }
```

Comment (Zhangxunmt) [reply plugin/src/test/java/org/opensearch/ml/rest/RestMLDeleteModelActionIT.java]: Yep, updated to use ExpectedException.

File: plugin/src/test/java/org/opensearch/ml/rest/RestMLGetModelActionTests.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.opensearch.common.Strings;
+import org.opensearch.rest.RestHandler;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.test.OpenSearchTestCase;
+
+public class RestMLGetModelActionTests extends OpenSearchTestCase {
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
```
Comment (ylwu-amzn) [inline plugin/src/test/java/org/opensearch/ml/rest/RestMLGetModelActionTests.java:21]: Minor: This is not being used. 
You can keep the commit history, especially after first round of review. So review can review what changed only. That can save some time.

**Review by ylwu-amzn — APPROVED** [review]
LGTM.

**Pull request merged**